tests/components/sleepiq/test_sensor.py

```diff
@@ -1,67 +1,44 @@
 from tests.components.sleepiq.test_init import mock_responses
 
+CONFIG = {"username": "foo", "password": "bar"}
 
-class TestSleepIQSensorSetup(unittest.TestCase):
-    """Tests the SleepIQ Sensor platform."""
-
-    DEVICES = []
-
-    def add_entities(self, devices):
-        """Mock add devices."""
-        for device in devices:
-            self.DEVICES.append(device)
-
-    def setUp(self):
-        """Initialize values for this testcase class."""
-        self.hass = get_test_home_assistant()
-        self.username = "foo"
-        self.password = "bar"
-        self.config = {"username": self.username, "password": self.password}
-        self.DEVICES = []
-        self.addCleanup(self.tear_down_cleanup)
 
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
+async def test_setup(hass, requests_mock):
+    """Test for successfully setting up the SleepIQ platform."""
+    mock_responses(requests_mock)
 
-    @requests_mock.Mocker()
-    def test_setup(self, mock):
-        """Test for successfully setting up the SleepIQ platform."""
-        mock_responses(mock)
+    assert await async_setup_component(hass, "sleepiq", {"sleepiq": CONFIG})
 
-        assert setup_component(self.hass, "sleepiq", {"sleepiq": self.config})
+    device_mock = MagicMock()
+    sleepiq.setup_platform(hass, CONFIG, device_mock, MagicMock())
+    devices = device_mock.call_args[0][0]
+    assert 2 == len(devices)
 
-        sleepiq.setup_platform(self.hass, self.config, self.add_entities, MagicMock())
-        assert 2 == len(self.DEVICES)
+    left_side = devices[1]
+    assert "SleepNumber ILE Test1 SleepNumber" == left_side.name
+    assert 40 == left_side.state
 
-        left_side = self.DEVICES[1]
-        assert "SleepNumber ILE Test1 SleepNumber" == left_side.name
-        assert 40 == left_side.state
+    right_side = devices[0]
+    assert "SleepNumber ILE Test2 SleepNumber" == right_side.name
+    assert 80 == right_side.state
 
-        right_side = self.DEVICES[0]
-        assert "SleepNumber ILE Test2 SleepNumber" == right_side.name
-        assert 80 == right_side.state
 
-    @requests_mock.Mocker()
-    def test_setup_sigle(self, mock):
-        """Test for successfully setting up the SleepIQ platform."""
-        mock_responses(mock, single=True)
+async def test_setup_sigle(hass, requests_mock):
+    """Test for successfully setting up the SleepIQ platform."""
+    mock_responses(requests_mock, single=True)
 
-        assert setup_component(self.hass, "sleepiq", {"sleepiq": self.config})
+    assert await async_setup_component(hass, "sleepiq", {"sleepiq": CONFIG})
 
-        sleepiq.setup_platform(self.hass, self.config, self.add_entities, MagicMock())
-        assert 1 == len(self.DEVICES)
+    device_mock = MagicMock()
+    sleepiq.setup_platform(hass, CONFIG, device_mock, MagicMock())
+    devices = device_mock.call_args[0][0]
+    assert 1 == len(devices)
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)